packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -469,22 +472,66 @@ class FocusNode with DiagnosticableTreeMixin, ChangeNotifier {
   ///  * [FocusTraversalPolicy], a class that can be extended to describe a
   ///    traversal policy.
   bool get canRequestFocus {
-    final FocusScopeNode scope = enclosingScope;
-    return _canRequestFocus && (scope == null || scope.canRequestFocus);
+    if (!_canRequestFocus) {
+      return false;
+    }
+    for (final FocusNode ancestor in ancestors) {
+      if (ancestor.overrideFocusability && !ancestor._canRequestFocus) {
+        return false;
+      }
+    }
+    return true;
   }
 
   bool _canRequestFocus;
   @mustCallSuper
   set canRequestFocus(bool value) {
     if (value != _canRequestFocus) {
-      if (!value) {
+      // Have to set this first before unfocusing, since it checks this to cull
+      // unfocusable, previously-focused children.
+      _canRequestFocus = value;
+      if (hasFocus && !value) {
         unfocus(disposition: UnfocusDisposition.previouslyFocusedChild);
       }
-      _canRequestFocus = value;
       _manager?._markPropertiesChanged(this);
     }
   }
 
+  /// If true, setting [canRequestFocus] on this focus node to false will
+  /// disable focus for its descendants.
+  ///
+  /// Defaults to false.  Set to true if you want this node's descendants to
+  /// be unfocusable when this node's [canRequestFocus] is false.
+  ///
+  /// If any descendants are focused when this is set and [canRequestFocus] is
+  /// false, they will be unfocused. When `overrideFocusability` is set to false
+  /// again, they will not be refocused, although they will be able to accept
+  /// focus again.
+  ///
+  /// See also:
+  ///
+  ///  * [ExcludeFocus], a widget that uses this property to conditionally
+  ///    exclude focus for a subtree.
+  ///  * [Focus], a widget that exposes this setting as a parameter.
+  ///  * [FocusTraversalGroup], a widget used to group together and configure
+  ///    the focus traversal policy for a widget subtree that also has an
+  ///    `excludeFocus` parameter that prevents its children from being focused.
```

Comment:
nit: replace with template use {flutter.widgets.Focus.overrideFocusability}

Reply:
No, this text refers to "this node", and the other refers to "this widget", which are important distinctions for the audience reading them.